code/game/gamemodes/cult/runes.dm

```diff
@@ -1,1053 +1,983 @@
-		if (istype(user, /mob/living))
-			user.take_overall_damage(5, 0)
+	else if(istype(I, /obj/item/weapon/nullrod))
+		user << "<span class='notice'>You disrupt the magic with [I].</span>"
 		qdel(src)
-	for(var/mob/living/C in orange(1,src))
-		if(iscultist(C) && !C.stat)
-			culcount++
-	if(user.loc==src.loc)
-		return fizzle(user)
-	if(culcount>=1)
-		user.say("Sas[pick("'","`")]so c'arta forbici tarem!")
-		user.visible_message("<span class='danger'>You feel air moving from the rune - like as it was swapped with somewhere else.</span>", \
-		"<span class='danger'>You feel air moving from the rune - like as it was swapped with somewhere else.</span>", \
-		"<span class='danger'>You smell ozone.</span>")
-		for(var/obj/O in src.loc)
-			if(!O.anchored)
-				O.loc = IP.loc
-		for(var/mob/M in src.loc)
-			M.loc = IP.loc
 		return
+	return
 
-	return fizzle(user)
-
-
-///Summon Tome: Summons a completely untranslated tome.
-
-
-/obj/effect/rune/proc/tomesummon()
-	if(istype(src,/obj/effect/rune))
-		usr.say("N[pick("'","`")]ath reth sh'yro eth d'raggathnor!")
+/obj/effect/rune/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		user << "<span class='warning'>You aren't able to understand the words of [src].</span>"
+		return
+	if(can_invoke(user))
+		invoke(user)
 	else
-		usr.whisper("N[pick("'","`")]ath reth sh'yro eth d'raggathnor!")
-	usr.visible_message("<span class='danger'>Rune disappears with a flash of red light, and in its place now a book lies.</span>", \
-	"<span class='danger'>You are blinded by the flash of red light! After you're able to see again, you see that now instead of the rune there's a book.</span>", \
-	"<span class='italics'>You hear a pop and smell ozone.</span>")
-	if(istype(src,/obj/effect/rune))
-		new /obj/item/weapon/tome(src.loc)
+		fail_invoke(user)
+
+/*
+
+There are a few different procs each rune runs through when a cultist activates it.
+can_invoke() is called when a cultist activates the rune with an empty hand. If there are multiple cultists, this rune determines if the required amount is nearby.
+invoke() is the rune's actual effects.
+fail_invoke() is called when the rune fails, via not enough people around or otherwise. Typically this just has a generic 'fizzle' effect.
+structure_check() searches for nearby cultist structures required for the invocation. Proper structures are pylons, forges, archives, and altars.
+
+*/
+
+/obj/effect/rune/proc/can_invoke(var/mob/living/user)
+	//This proc determines if the rune can be invoked at the time. If there are multiple required cultists, it will find all nearby cultists.
+	if(!structure_check(req_pylons, req_forges, req_archives, req_altars))
+		return 0
+	if(!keyword && req_keyword)
+		return 0
+	if(req_cultists <= 1)
+		if(invocation)
+			user.say(invocation)
+		return 1
 	else
-		new /obj/item/weapon/tome(usr.loc)
-	qdel(src)
-	return
+		var/cultists_in_range = 0
+		for(var/mob/living/L in orange(1, src))
+			if(iscultist(L))

```

Comment:
Will fix all these, but I'll leave the ability to invoke if weakened because the cultists nearby are chanting. Only the invoker is actually touching the rune.